### PR TITLE
Consumer FP registration test

### PR DESCRIPTION
**Pull request by maurolacy**

### CODE REVIEW

File: finality-provider/service/fp_manager.go [hunk 1]
```diff
@@ -293,6 +293,20 @@ func (fpm *FinalityProviderManager) ListFinalityProviderInstances() []*FinalityP
 	return fpisList
 }
 
+func (fpm *FinalityProviderManager) ListFinalityProviderInstancesForChain(chainID string) []*FinalityProviderInstance {
+	fpm.mu.Lock()
+	defer fpm.mu.Unlock()
+
+	fpisList := make([]*FinalityProviderInstance, 0, len(fpm.fpis))
+	for _, fpi := range fpm.fpis {
+		if string(fpi.GetChainID()) == chainID {
```
Comment (SebastianElvis) [inline finality-provider/service/fp_manager.go:302]: I saw `GetChainID()` is defined as
```go
func (fp *FinalityProviderInstance) GetChainID() []byte {
	return []byte(fp.state.getStoreFinalityProvider().ChainID)
}
```
so probably we could make it return `string`?

**Review by SebastianElvis — APPROVED** [review]

File: clientcontroller/babylon.go [hunk 2]
```diff
@@ -586,3 +587,20 @@ func (bc *BabylonController) SubmitCovenantSigs(
 
 	return &types.TxResponse{TxHash: res.TxHash, Events: res.Events}, nil
 }
+
+// Test methods for e2e testing
+func (bc *BabylonController) RegisterConsumerChain(id, name, description string) (*types.TxResponse, error) {
```
Comment (gitferry) [inline clientcontroller/babylon.go:592]: Would it be part of client controller interface?

File: itest/e2e_test.go [hunk 3]
```diff
@@ -203,3 +203,18 @@ func TestFastSync(t *testing.T) {
...
+func TestConsumerFinalityProviderRegistration(t *testing.T) {
+	tm, _ := StartManagerWithFinalityProvider(t, 1)
+	defer tm.Stop(t)
+
+	consumerChainID := "consumer-chain-test-1"
+
+	_, err := tm.BBNClient.RegisterConsumerChain(consumerChainID, "Consumer chain 1 (test)", "Test Consumer Chain 1")
+	if err != nil {
+		t.Fatalf("failed to register consumer chain: %v", err)
+	}
```
Comment (gitferry) [inline itest/e2e_test.go:217]:
```suggestion
	require.NoError(t, err)
```

File: itest/e2e_test.go [hunk 4]
```diff
@@ -203,3 +203,18 @@ func TestFastSync(t *testing.T) {
...
+
+	tm.CreateFinalityProvidersForChain(t, consumerChainID, 1)
```
Comment (gitferry) [inline itest/e2e_test.go:219]: I think this test also needs to check whether the chain and fps are registered on the Babylon side. For that, we might need some queries

**Review by gitferry — COMMENTED** [review]
Nice work! One comment is that we probably need queries to ask Babylon for registration status. Local info is only half the story

File: itest/e2e_test.go (hunk 4 above)
Comment (maurolacy) [reply itest/e2e_test.go:217]: Queries are there. Will try to map them and use them, in a follow up.

**Pull request merged**